src/View/ViewableData_Customised.php

```diff
@@ -59,19 +59,26 @@ public function hasMethod($method)
         return $this->customised->hasMethod($method) || $this->original->hasMethod($method);
     }
 
-    public function cachedCall($field, $arguments = null, $identifier = null)
+    public function cachedCall($fieldName, $arguments = null, $identifier = null)
     {
-        if ($this->customised->hasMethod($field) || $this->customised->hasField($field)) {
-            return $this->customised->cachedCall($field, $arguments, $identifier);
+        if ($this->customisedHas($fieldName)) {
+            return $this->customised->cachedCall($fieldName, $arguments, $identifier);
         }
-        return $this->original->cachedCall($field, $arguments, $identifier);
+        return $this->original->cachedCall($fieldName, $arguments, $identifier);
     }
 
     public function obj($fieldName, $arguments = null, $cache = false, $cacheName = null)
     {
-        if ($this->customised->hasField($fieldName) || $this->customised->hasMethod($fieldName)) {
+        if ($this->customisedHas($fieldName)) {
             return $this->customised->obj($fieldName, $arguments, $cache, $cacheName);
         }
         return $this->original->obj($fieldName, $arguments, $cache, $cacheName);
     }
+
+    private function customisedHas(string $fieldName): bool
+    {
+        return property_exists($this->customised, $fieldName) ||
```

Comment:
Why explicitly check `property_exists` here? Doesn't `hasField()` check that already?

Reply:
`DataObject::hasField()` is irrelevant, since `DataObject` is not a subclass of `ViewableData_Customised`... that said, I _suppose_ someone could inject a custom copy of `ViewableData_Customised` somewhere which doesn't include the `property_exists` check in `hasField()`. Unlikely, but this protects against it with no downside that I can see so I'm okay with it.

Reply:
the property_exists() check is on $this->customised, not $this.  $this->customed is a ViewableData, NOT a ViewableData_Customised